cou/apps/auxiliary.py

```diff
@@ -183,6 +183,87 @@ class RabbitMQServer(AuxiliaryApplication):
     # COU changes to 3.9 if the channel is set to 3.8
     multiple_channels = True
 
+    def pre_upgrade_steps(
+        self, target: OpenStackRelease, units: Optional[list[Unit]]
+    ) -> list[PreUpgradeStep]:
+        """Pre Upgrade steps planning.
+
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
+        :param units: Units to generate upgrade plan
+        :type units: Optional[list[Unit]]
+        :return: List of pre upgrade steps.
+        :rtype: list[PreUpgradeStep]
+        """
+        steps = super().pre_upgrade_steps(target, units)
+        if self.config["enable-auto-restarts"].get("value") is False:
+            action_name = "run-deferred-hooks"
+            units_to_run_action = self.units.values() if units is None else units
+            steps += [
+                PreUpgradeStep(
+                    description="Auto restarts is disabled, will"
+                    f" execute run-deferred-hooks for unit: '{unit.name}'",
+                    coro=self.model.run_action(unit.name, action_name, raise_on_failure=True),
+                )
+                for unit in units_to_run_action
+            ]
+            steps += [
+                PreUpgradeStep(
```

Comment:
Why do you think this is necessary here? Maybe it's good to have it here, but definitely no with `apps=None`, so waiting for whole model.

Reply:
that's because the `RabbitmqServer` class default to `wait_for_model=True`

Reply:
I think maybe it's better to wait for it to settle, since we are running an action

Reply:
I think it fine, not sure if really needed but definitely fine.